### PR TITLE
Summarize step changes for metrics

### LINKED ISSUE #47148: [FE] Summarize step copy changes

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/47147
Closes https://github.com/metabase/metabase/issues/47148

Changes for metrics:
- "Summarize" step title becomes "Formula"
- "Default time dimension" title for breakouts
- Cannot remove the summarize step

Implementation details:
- Allows a custom `StepHeader` component for a step. Because of the layout issues, we need to have the breakout title in 2 places. This is extensively tested.
- While not used in `SummarizeStepHeader`, `steps` do not allow reverting the summarize step.

Screenshot 2024-08-23 at 21 54 54
Screenshot 2024-08-23 at 21 55 05
Screenshot 2024-08-23 at 21 55 34

### CODE REVIEW

File: frontend/src/metabase/querying/notebook/components/NotebookStep/NotebookStep.tsx [hunk 1]
```diff
@@ -115,57 +108,44 @@ function NotebookStep({
...
+                hasPreviewButton={hasPreviewButton}
+                onClick={openPreview}
+                data-testid="step-preview-button"
               />
-            </IconButtonWrapper>
+            </StepButtonContainer>
           )}
-        </StepHeader>
-
-        {NotebookStepComponent && (
```
Comment (ranquild) [inline frontend/src/metabase/querying/notebook/components/NotebookStep/NotebookStep.tsx:139]: Removed the condition, everything else is whitespace

File: frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStepHeader/SummarizeStepHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        title={title}
+        color={color}
+        canRevert={canRevert}
+        onRevert={onRevert}
+      />
+    );
+  }
+
+  return (
+    <Flex c={color} fw="bold" mb="sm" gap="md">
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStepHeader/SummarizeStepHeader.tsx:29]: The header uses 1rem gap and the blocks below use 0.5rem gap.

image

File: frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx [hunk 3]
```diff
@@ -22,8 +24,13 @@ export function SummarizeStep({
           {...props}
         />
       </Box>
-      <Box c={color} fw="bold">{t`by`}</Box>
+      {!isMetric && <Box c={color} fw="bold">{t`by`}</Box>}
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx:27]: With "Default time dimension" left-aligned on mobile, the center-aligned "by" looks weird right now.
Let's style them both in the same way. IMO left-aligned looks better (and all other labels are aligned this way).


image

File: frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx [hunk 4]
```diff
@@ -22,8 +24,13 @@ export function SummarizeStep({
...
       <Box w={{ base: "100%", md: "50%" }}>
+        {isMetric && (
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx:29]: Why does this element have a different parent than the one with `"by"`?

**Review by kamilmielnik — APPROVED** [review]
Works as expected. Minor visual feedback inline.

File: frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStep.tsx]: Fixed

File: frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStepHeader/SummarizeStepHeader.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/querying/notebook/components/SummarizeStep/SummarizeStepHeader/SummarizeStepHeader.tsx:29]: Fixed

**Pull request merged**